10-network/lab/solution/src/bg/sofia/uni/fmi/mjt/wish/list/WishListServer.java

```diff
@@ -0,0 +1,178 @@
+
+                while (keyIterator.hasNext()) {
+                    SelectionKey key = keyIterator.next();
+                    if (key.isReadable()) {
+                        SocketChannel socketChannel = (SocketChannel) key.channel();
+
+                        messageBuffer.clear();
+                        int r = socketChannel.read(messageBuffer);
+                        if (r <= 0) {
+                            System.out.println("Nothing to read, closing channel");
+                            socketChannel.close();
+                            break;
+                        }
+
+                        handleKeyIsReadable(key, messageBuffer);
+                    } else if (key.isAcceptable()) {
+                        handleKeyIsAcceptable(selector, key);
+                    }
+
+                    keyIterator.remove();
+                }
+            }
+        } catch (IOException e) {
+            System.err.println("There is a problem with the server socket: " + e.getMessage());
+            System.err.println(e);
+        }
+
+        System.out.println("Server stopped");
+    }
+
+    public void stop() {
+        isStarted = false;
+    }
+
+    private void handleKeyIsReadable(SelectionKey key, ByteBuffer buffer) throws IOException {
+        SocketChannel socketChannel = (SocketChannel) key.channel();
+
+        buffer.flip();
+        String message = new String(buffer.array(), 0, buffer.limit()).trim();
+
+        System.out.println("Message [" + message + "] received from client " + socketChannel.getRemoteAddress());
+
+        String command = message.split(" ")[0];
+        String arguments = message.substring(message.indexOf(" ") + 1);
+
+        String response = null;
+        switch (command) {
+        case "post-wish":
+            response = postWish(arguments);
+            break;
+        case "get-wish":
+            response = getWish();
+            break;
+        case "disconnect":
+            disconnect(key);
+            break;
+        default:
+            response = "[ Unknown command ]";
+        }
+
+        if (response != null) {
+            System.out.println("Sending response to client: " + response);
+            response += System.lineSeparator();
+            buffer.clear();
+            buffer.put(response.getBytes());
+            buffer.flip();
+            socketChannel.write(buffer);
+        }
+    }
+
+    private void handleKeyIsAcceptable(Selector selector, SelectionKey key) throws IOException {
+        ServerSocketChannel sockChannel = (ServerSocketChannel) key.channel();
+        SocketChannel accept = sockChannel.accept();
+        accept.configureBlocking(false);
+        accept.register(selector, SelectionKey.OP_READ);
+
+        System.out.println("Connection accepted from client " + accept.getRemoteAddress());
+    }
+
+    private String postWish(String arguments) {
+        String[] argumentss = arguments.split("\\s+", 2);

```

Comment:
Maybe rename "argumentss" to e.g. "argumentsArray"

Reply:
Done.